.github/workflows/update.yml

```diff
@@ -113,6 +121,38 @@ jobs:
             echo "EOF" >> "$GITHUB_OUTPUT"
           fi
 
+      - name: Update dev flake.lock
+        id: dev_flake_lock
+        if: inputs.dev_lock || github.event_name == 'schedule'
+        run: |
+          root_nixpkgs=$(nix eval -f . 'inputs.nixpkgs.rev')
+          old=$(git show --no-patch --format=%h)
+          nix flake update --commit-lock-file \
+              --override-input nixpkgs "github:NixOS/nixpkgs/$root_nixpkgs" \
+              --flake './flake/dev'
+          new=$(git show --no-patch --format=%h)
+          if [ "$old" != "$new" ]; then
+            echo "body<<EOF" >> "$GITHUB_OUTPUT"
+            git show --no-patch --format=%b >> "$GITHUB_OUTPUT"
+            echo "EOF" >> "$GITHUB_OUTPUT"
+          fi
+
+      - name: Update docs flake.lock
+        id: docs_flake_lock
+        if: inputs.docs_lock || github.event_name == 'schedule'
+        run: |
+          root_nixpkgs=$(nix eval -f . 'inputs.nixpkgs.rev')
+          old=$(git show --no-patch --format=%h)
+          nix flake update --commit-lock-file \
+              --override-input nixpkgs "github:NixOS/nixpkgs/$root_nixpkgs" \
+              --flake './docs'
```

Comment:
This is _kinda_ awkward; we can't easily re-use the nixpkgs input from the root module because flake's don't work very well when referencing other flakes with relative paths. https://github.com/NixOS/nix/pull/10089 improves things, but we can't take advantage of it until nix 2.26.

For now we're stuck manually trying to keep nixpkgs in sync across lock files. We can do that fairly easily here using `--override-input`, but we have to remember to do the same if we ever update lock files manually.

That said, we should prefer using the ci workflow to update nixpkgs _anyway_, because the CI will also check if the generated files need to change.